lib/src/query_component.dart

```diff
@@ -93,26 +100,20 @@ class QueryComponent {
       showAlert = false;
 
       // Uses DV360 public APIs to fetch entity data.
-      List<InsertionOrder> insertionOrders =
-          await _queryAndParseInsertionOrderEntityData();
-
-      // Early return if the query returns no insertion orders.
-      if (insertionOrders.isEmpty) return;
-
-      // If query type is not by insertion order, filters out insertion orders
-      // that are not in-flight.
-      insertionOrders = queryType == QueryType.byInsertionOrder
-          ? insertionOrders
-          : _filterInFlightInsertionOrders(insertionOrders);
-
-      // Gets the earliest start date from the list of insertion orders.
-      final minStartDate = _getMinStartDate(insertionOrders);
+      Future<List<InsertionOrder>> publicApiResponse =
+          _queryAndParseInsertionOrderEntityData();
+      apiRequestFutures.add(publicApiResponse);
 
       // Uses DBM reporting APIs to get spent (currency based or
-      // impression based) within time window [minStartDate, Now].
-      // spendingMap has a [DailySpend] for each insertion order.
-      final spendingMap =
-          await _queryAndParseSpentData(minStartDate, DateTime.now());
+      // impression based) within time window [reportStartDate, Now].
+      Future<Multimap<String, InsertionOrderDailySpend>> reportingApiResponse =
+          _queryAndParseSpentData(reportStartDate, DateTime.now());
+      apiRequestFutures.add(reportingApiResponse);
+
+      // Waits for all requests to finish.
+      final responseList = await Future.wait(apiRequestFutures);
```

Comment:
do you have to wait for both to finish to start parsing one or the other?
is the time savings worth the complexity of parsing one before the other finishes?

Reply:
publicApiResponse and reportingApiResponse are already parsed. So the future is resolved when both requests respond back and finish parsing. In the way, I think what I'm doing here is also parsing the two responses in two separate isolates (Dart version of threads?) 

I think it keeps a good balance? i.e. public api takes longer to response but less time to parse, reporting api takes less time to response (on average) but longer time to parse. 

And then the joining of spending data to IOs happen at the last.